spark-icons/src/main/kotlin/com/adevinta/spark/icons/SparkIcon.kt

```diff
@@ -0,0 +1,637 @@
+        public object AttachFile : Share(R.drawable.spark_icons_attachment)
+        public object Facebook : Share(R.drawable.spark_icons_facebook)
+        public object Messenger : Share(R.drawable.spark_icons_messenger)
+        public object Pinterest : Share(R.drawable.spark_icons_pinterest)
+        public object Instagram : Share(R.drawable.spark_icons_instagram)
+        public object Goto : Share(R.drawable.spark_icons_goto)
+        public object Link : Share(R.drawable.spark_icons_link)
+        public object ShareDefault : Share(R.drawable.spark_icons_share)
+        public object ShareArrow : Share(R.drawable.spark_icons_share_arrow)
+        public object Twitter : Share(R.drawable.spark_icons_twitter)
+        public object WhatsApp : Share(R.drawable.spark_icons_whatsapp)
+        public object Upload : Share(R.drawable.spark_icons_upload)
+        public object Download : Share(R.drawable.spark_icons_share_download)
+    }
+
+    public sealed class Toggles(@DrawableRes override val drawableId: Int) : SparkIcon(drawableId) {
+        public object Add : Toggles(R.drawable.spark_icons_add)
+
+        public sealed class Check(@DrawableRes override val drawableId: Int) : Toggles(drawableId) {
+            public object Default : Check(R.drawable.spark_icons_check)
+            public object Outline : Check(R.drawable.spark_icons_check_outline)
+            public object Simple : Toggles(R.drawable.spark_icons_check_simple)
+            public object Double : Toggles(R.drawable.spark_icons_check_double)
+        }
+    }
+
+    public sealed class User(@DrawableRes override val drawableId: Int) : SparkIcon(drawableId) {
+        public object Pro : User(R.drawable.spark_icons_profile_pro2)
+        public object Default : User(R.drawable.spark_icons_user)
+        public object Outline : User(R.drawable.spark_icons_user_outline)
+        public sealed class Group(@DrawableRes override val drawableId: Int) : User(drawableId) {
+            public object Default : Group(R.drawable.spark_icons_user_group)
+            public object Outline : Group(R.drawable.spark_icons_user_group_outline)
+        }
+
+        public object Verified : User(R.drawable.spark_icons_verified)
+        public object Avatar : User(R.drawable.spark_icons_avatar_part)
+        public object Store : User(R.drawable.spark_icons_profile_pro1_grey)
+    }
+
+    public sealed class Value(@DrawableRes override val drawableId: Int) : SparkIcon(drawableId) {
+        public object Minus : Value(R.drawable.spark_icons_minus_outline)
+
+        public sealed class More(@DrawableRes override val drawableId: Int) : Value(drawableId) {
+            public object Default : More(R.drawable.spark_icons_more)
+            public object Outline : More(R.drawable.spark_icons_more_outline)
+        }
+    }
+
+    public abstract class Logo(@DrawableRes override val drawableId: Int) : SparkIcon(drawableId)
```

Comment:
IMO This should not be call Logo, but something more generic. For instance `Custom`. WDYT?

```suggestion
    public abstract class Custom(@DrawableRes override val drawableId: Int) : SparkIcon(drawableId)
```